pinot-core/src/main/java/org/apache/pinot/core/transport/ServerChannels.java

```diff
@@ -139,22 +143,33 @@ private void attachSSLHandler(SocketChannel ch) {
       }
     }
 
-    synchronized void sendRequest(String rawTableName, AsyncQueryResponse asyncQueryResponse,
-        ServerRoutingInstance serverRoutingInstance, InstanceRequest instanceRequest)
+    private void sendRequestOrTimeOut(String rawTableName, AsyncQueryResponse asyncQueryResponse,
+        ServerRoutingInstance serverRoutingInstance, byte[] requestBytes, long timeoutMs)
+        throws Exception {
+      _channelLock.lock();
+      if (_channelLock.tryLock(timeoutMs, TimeUnit.MILLISECONDS)) {
```

Comment:
Line 149 looks unintentional

Reply:
Oh, thanks for pointing out that. That explains why integration test timed out 😂